### PR TITLE
Add TotalTx to block header

### PR DESCRIPTION
**Pull request by ethanfrey**
Addresses one part of issue #952

 - Add extra field to header
 - Make NumTx also an int64 for consistency.
 - Update state to keep track of this info.
 - Change function args as needed.

### CODE REVIEW

File: lite/helpers.go [hunk 1]
```diff
@@ -112,10 +112,11 @@ func genHeader(chainID string, height int64, txs types.Txs,
...
 
 	return &types.Header{
-		ChainID: chainID,
-		Height:  height,
-		Time:    time.Now(),
-		NumTxs:  len(txs),
+		ChainID:  chainID,
+		Height:   height,
+		Time:     time.Now(),
+		NumTxs:   int64(len(txs)),
```
Comment (melekes) [inline lite/helpers.go]: should this be int64? int32 should be more than enough

Comment (melekes) [reply lite/helpers.go]: or it's to avoid int64(..) cast?

File: consensus/mempool_test.go [hunk 2]
```diff
@@ -108,7 +108,8 @@ func TestTxConcurrentWithCommit(t *testing.T) {
 	for nTxs := 0; nTxs < NTxs; {
```
Comment (melekes) [inline consensus/mempool_test.go]: should we make nTxs int64 like `nTxs := int64(0)`

File: lite/helpers.go (hunk 1 above)
Comment (ethanfrey) [reply lite/helpers.go]: yup, so it is the same type as TotalTxs. No need to have them different types, just confusion when adding

File: consensus/mempool_test.go (hunk 2 above)
Comment (ethanfrey) [reply consensus/mempool_test.go]: I could, but then I would have to change types here as well:
```
	NTxs := 10000
	go deliverTxsRange(cs, 0, NTxs)
```

simpler to cast here once

**Comment by ethanfrey** [comment]
Okay, wrote some block tests that were non-existent. Should work now.

**Comment by ebuchman** [comment]
Great. Thanks for the test.

I wonder if the MakeBlock/ValidateBlock could be better structured, since the args list is a little out of control :P. For instance maybe just take a block and it's parent. Might ripple through the State in ways we don't like though. Just a thought.

**Comment by ethanfrey** [comment]
Yeah, I just wanted to get some test coverage at all.
Probably could be cleaned up, but I don't know all usages of it. Will keep it in mind as I see these functions in use.

**Pull request merged**